xbmc/cores/VideoPlayer/VideoRenderers/windows/RendererDXVA.cpp

```diff
@@ -72,13 +54,26 @@ void CRendererDXVA::GetWeight(std::map<RenderMethod, int>& weights, const VideoP
                  DX::DXGIFormatToString(dxgi_format));
       return;
     }
+  }
 
-    if (enumerator.SupportedConversions(SupportedConversionsArgs{picture, systemUsesHDR}).empty())
-    {
-      CLog::LogF(LOGWARNING, "DXVA will not be used.");
-      return;
-    }
+  CEnumeratorHD enumerator;
+  enumerator.Open(picture.iWidth, picture.iHeight, dxgi_format);
+
+  const bool renderAsHDR =
+      IntendToRenderAsHDR(static_cast<AVColorPrimaries>(picture.color_primaries),
+                          static_cast<AVColorTransferCharacteristic>(picture.color_transfer));
+  if (enumerator.SupportedConversions({picture, renderAsHDR}).empty())
```

Comment:
A blank line here will help improve readability

Reply:
It can be compacted yes, looked more readable with temp variable at some point but it's ok again. Changed.

About VideoPicture vs discrete parameters: it's a philosophical decision I guess and I could go either way.
When using only one or two members, I don't think it's worth using the VideoPicture, and what the function uses is clearer from the function signature. If it was 3 or more, then sure, too many parameters are less readable.

About the casts: a VideoPicture parameter moves the casts inside IntendToRender, it's not like they go away.
There could be a debate about defining better types inside VideoPicture: I understand it's not a great idea to use types/enums from an external library, but simply changing the type to unsigned int in VideoPicture and keeping the same meaning for each value is also not great.
I've been using the AV enums because they provide a bit more safety than simple int with no clear defined meaning, but I don't like it very much.

To do a better job, we'd probably need Kodi enums to represent the primaries, transfer, etc and conversion functions in one place.
Then we could avoid casts in many places, like here and a few others. Touches all platforms though.
What do you think?